src/api.raml

```diff
@@ -506,29 +506,65 @@ traits:
                             Article version number.
                         type: integer
                         minimum: 1
-                get:
-                    description: |
-                        Get an article.
+                put:
+                    description: Ingest an article.
+                    queryParameters:
```

Comment:
never seen query string parameters on a `PUT`; analyzing what they do:

`force` allows the `PUT` to have different results depending on the existing state, probably silent correction vs new ingest?
`dry-run` doesn't change the state but only validates everything will be fine when not used.

Both use cases seem to be served by different resources?
- `/article/:id/versions/:v:/
- `/article/:id/versions/:v:/dry-run
but it would be helpful to elaborate on their intended meaning to decide, especially `force`

Reply:
`force` sounds like [conditional requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Conditional_requests)?

What does `dry-run` do that validating against the schema wouldn't?

Reply:
> /article/:id/versions/:v:/dry-run

it would be `/article/:id/verisons/:v?dry-run=true`

> What does dry-run do that validating against the schema wouldn't?

1. lax contains business rules about what can be stored. it protects against stupid things, like trying to ingest or publish a version 2 when there is no version 1.

2. the article-json isn't complete coming out of bot-lax, there is information that is required that can't be known without knowing the complete article history. Validation occurs prior to lax with placeholders, dummy values that are then removed in lax.

3. lax also accumulates article 'fragments', pushed by journal-cms, and what is served up is the merged and validated result of all of these fragments.

4. bot-lax has an API that is used by Exeter to test articles are correct. the xml is transformed -> sent to lax -> validated and then the result is rolled back.

Reply:
1 to 3 are all good concerns, whereas 4 seems to be the only scenario that involves calling the API with `dry-run`. That seems to me a different endpoint with a different implementation, e.g. `/article/:id/tentative-versions/:v`